connectors/sources/sharepoint.py

```diff
@@ -44,15 +44,18 @@
 LIST_ITEM = "list_item"
 ATTACHMENT_DATA = "attachment_data"
 DOCUMENT_LIBRARY = "document_library"
+LIST_BY_TITLE = "list_by_title"
+TIKA_SUPPORTED_FILETYPES.append(".aspx")
```

Comment:
Why do we want to send the `aspx` file to Apache Tika?

Reply:
We are extracting the content of aspx(i.e site pages) only for sharepoint connector and if we directly update the list of TIKA_SUPPORTED_FILETYPES in `utils.py` then this will be applicable for all the connectors. That's why we are keeping it specific only for sharepoint only.

Reply:
Rendered content is HTML not the .NET source code right? So if you want to extract raw text from HTML then the file is `.html` right? 

Also, could you please point it me where exactly the connector sends `.aspx` files to tika?

Reply:
> We are extracting the content of aspx(i.e site pages) only for sharepoint connector and if we directly update the list of TIKA_SUPPORTED_FILETYPES in utils.py then this will be applicable for all the connectors. That's why we are keeping it specific only for sharepoint only.

I think you're doing exposing this for all connectors anyway - you're not creating a new variable for `TIKA_SUPPORTED_FILETYPES`, you're changing the value of the global variable. As soon as this python file is loaded,  `.aspx` will be in the list of values for any connector that references this global variable.

```
>>> from connectors.utils import TIKA_SUPPORTED_FILETYPES
>>> TIKA_SUPPORTED_FILETYPES
['.txt', '.py', '.rst', '.html', '.markdown', '.json', '.xml', '.csv', '.md', '.ppt', '.rtf', '.docx', '.odt', '.xls', '.xlsx', '.rb', '.paper', '.sh', '.pptx', '.pdf', '.doc']
>>> import connectors.sources.sharepoint
>>> TIKA_SUPPORTED_FILETYPES
['.txt', '.py', '.rst', '.html', '.markdown', '.json', '.xml', '.csv', '.md', '.ppt', '.rtf', '.docx', '.odt', '.xls', '.xlsx', '.rb', '.paper', '.sh', '.pptx', '.pdf', '.doc', '.aspx']
>>> 
```

Reply:
> Rendered content is HTML not the .NET source code right? So if you want to extract raw text from HTML, the file is `.html` right?

Yes the rendered conetnt is HTML in response but the extension in filename for web pages is `.aspx`. And in our connector we are verifying the supported filetype with there name so that's why we need to add it in `TIKA_SUPPORTED_FILETYPES`
> Also, could you please point me to where exactly the connector sends `.aspx` files to tika?

this will be send in `_attachment` field along with the condition that the value in this field must be a base64 encoded 
binary
for ref: https://crestdatasystems.slack.com/archives/C02AV5C8ZDE/p1682058879139289?thread_ts=1681798035.637829&cid=C02AV5C8ZDE
https://github.com/elastic/connectors-python/blob/main/connectors/sources/sharepoint.py#L248

Reply:
> > We are extracting the content of aspx(i.e site pages) only for sharepoint connector and if we directly update the list of TIKA_SUPPORTED_FILETYPES in utils.py then this will be applicable for all the connectors. That's why we are keeping it specific only for sharepoint only.
> 
> I think you're doing exposing this for all connectors anyway - you're not creating a new variable for `TIKA_SUPPORTED_FILETYPES`, you're changing the value of the global variable. As soon as this python file is loaded, `.aspx` will be in the list of values for any connector that references this global variable.
> 
> ```
> >>> from connectors.utils import TIKA_SUPPORTED_FILETYPES
> >>> TIKA_SUPPORTED_FILETYPES
> ['.txt', '.py', '.rst', '.html', '.markdown', '.json', '.xml', '.csv', '.md', '.ppt', '.rtf', '.docx', '.odt', '.xls', '.xlsx', '.rb', '.paper', '.sh', '.pptx', '.pdf', '.doc']
> >>> import connectors.sources.sharepoint
> >>> TIKA_SUPPORTED_FILETYPES
> ['.txt', '.py', '.rst', '.html', '.markdown', '.json', '.xml', '.csv', '.md', '.ppt', '.rtf', '.docx', '.odt', '.xls', '.xlsx', '.rb', '.paper', '.sh', '.pptx', '.pdf', '.doc', '.aspx']
> >>> 
> ```
I assumed that for now we can run single connector at a time, and in that case once we add connector_id in config.yml for running a specific connector and if the connector is sharepoint only then `aspx` will append to  `TIKA_SUPPORTED_FILETYPES` . for running a different connector if we change the connector_id then it will consider it as a new sync and `aspx` will be removed from the list.
But if you want to store this in a new variable I can do this change.

Reply:
Let's either add `.aspx` directly to https://github.com/elastic/connectors-python/blob/main/connectors/utils.py#L34-L56, or simply change the file name to be `.html` instead of `.aspx`. Either is fine. But it doesn't make sense to modify this global variable here.